### PR TITLE
VZ-6774,VZ-6773: Build pipeline improvements -  remove unnecessary full cluster snapshots, run analysis tool on bug-report

### PR DESCRIPTION
**Pull request by bhatpmk**
This PR improves the build pipelines by removing the calls to capture the cluster snapshot which are not required. For example, after the Verrazzano installation in ci/scripts/prepare_jenkins_at_environment.sh and ci/scripts/prepare_private_registry_jenkins_at_environment.sh. 
The code to capture the cluster dump is moved to a new script, which is called by the groovy function in the individual Jenkins scripts. 
A new parameter is added to capture the full cluster from test failures, when the pipeline defines required environment variables to capture cluster resources on failure. This is a boolean parameter, with default value false.

### CODE REVIEW

File: ci/scripts/capture_cluster_snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  echo "The script to capture the cluster resources ${CLUSTER_SNAPSHOT_SCRIPT} doesn't exist"
+  exit 1
+fi
+
+FULL_CLUSTER_DIR=${CLUSTER_SNAPSHOT_ROOT}/full-cluster
+BUG_REPORT_DIR=${CLUSTER_SNAPSHOT_ROOT}/bug-report
+BUG_REPORT_FILE="${BUG_REPORT_DIR}/${BUG_REPORT}"
+
+captureFullCluster
+captureBugReport
```
[the remark this answers is not in the document]
Comment (bhatpmk) [reply ci/scripts/capture_cluster_snapshot.sh:53]: Yes, we will always capture bug-report

File: ci/backup/JenkinsfileAlllBackupOKETest [hunk 2]
```diff
@@ -155,6 +155,8 @@ pipeline {
         OCI_OS_COMMIT_BUCKET="verrazzano-builds-by-commit"
         VZ_CLI_TARGZ="vz-linux-amd64.tar.gz"
 
+        // Environment variable for Verrazzano CLI executable
+        VZ_COMMAND="${GO_REPO_PATH}/vz"
```
Comment (agv426) [inline ci/backup/JenkinsfileAlllBackupOKETest:159]: We may want to have it pull down the CLI from object storage earlier than in the install for this job (other jobs we pull it down in the initial stage). Not sure if any captures will be attempted before it pulls it down where it is now

Comment (bhatpmk) [reply ci/backup/JenkinsfileAlllBackupOKETest:159]: Good catch. Updated the file to pull CLI to stage "Clean workspace and checkout"

**Review by agv426 — DISMISSED** [review]

**Review by Abhijithang — DISMISSED** [review]

**Review by anajosep — APPROVED** [review]

**Comment by bhatpmk** [comment]
VZ-6774
VZ-6773

**Pull request merged**